renderer/backend/metal/command_buffer_mtl.mm

```diff
@@ -55,8 +55,8 @@
 
 // TODO(dnfield): remove this declaration when we no longer need to build on
 // machines with lower SDK versions than 11.0.s
-#if !defined(MAC_OS_X_VERSION_11_0) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_11_0
-typedef enum MTLCommandEncoderErrorState : NSInteger {
+#if !defined(MAC_OS_VERSION_11_0) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_VERSION_11_0
```

Comment:
All Mac engine builders except for the `Mac Web Engine` are running Xcode 13 (which has `MAC_OS_VERSION_11_0` defined)
https://github.com/flutter/engine/blob/289368cadccbce88913b3ab279ed754d705740f9/.ci.yaml#L73

`Mac Web Engine` Xcode migration blocked on https://github.com/flutter/flutter/issues/99885
https://github.com/flutter/engine/blob/289368cadccbce88913b3ab279ed754d705740f9/.ci.yaml#L327

Reply:
Need a guard for the macOS 12 defines too

Reply:
So once Mac Web engine rolls we can remove the first block here - we won't be able to remove the second until we further upgrade.